### DISCUSSION TITLE
chapter_dynamic_programming/summary/

### DISCUSSION THREAD

**Discussion by giscus** [krahets/hello-algo#603] [Announcements]
# chapter_dynamic_programming/summary/

动画图解、一键运行的数据结构与算法教程

https://www.hello-algo.com/chapter_dynamic_programming/summary/

**Comment by lizujin-magictavern-com** [comment]
learned，over！

**Reply by Allen-Scai** [reply]
真的学习到很多，之前对于动态规划，最多只能是理解到01背包问题了。本书的作者用大量的用例和解释说明，让我一步步地学习和了解了动态规划相关的问题应该怎么分析。比如先考虑暴力搜索，接着尝试优化，紧接着分析问题是否`具有重叠子问题、最优子结构、无后效性`这三个特点，来看是否可以使用动态规划。最后就是动态规划解法以及对应的空间优化了，真的十分感谢本书的所有作者！

**Comment by Magicperformer** [comment]
真的受教了，之前对于零钱兑换和完全背包问题感到一筹莫展，完全无法理解，现在通过不停观察和应用类递归的动态规划算是初窥门径了，赞美本书的所有作者，谢谢你们讲的这么浅显易懂

**Comment by MambaFly** [comment]
wohoooooo！学完动态规划，总算是透彻了！感觉这种思想在很多地方都有运用，继续刷题！

**Comment by jungleford** [comment]
最值得反复回味的一章。
难的是选对方向，方向对了就成功了七成，剩下的都是套路。

**Comment by Xi-Row** [comment]
动态规划给我的感觉很像高中的数列递推和马尔科夫链，甚至可以说他们本质都是同一个东西，动态规划讲得太好了，空间优化太形象了！

**Comment by superfeimao** [comment]
感觉写的真好，没有废话，每一句话都有用！

**Comment by linjue2006** [comment]
对于初学者自学还是蛮有难度的，尤其是背包和编辑距离问题的空间优化方案。很多东西虽然说得很明白，但还是要慢慢自己琢磨思考

**Comment by DerrickA1st** [comment]
无后效性：当前状态一旦确定，未来的状态转移只依赖当前状态中已经包含的信息，而不依赖形成当前状态的具体历史过程
